lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -218,6 +218,30 @@ public function setTemplateOptions($templateOptions)
     }
 
     /**
+     * Creates a Zend_Mime_Part attachment and attaches it to the current message scope.
+     * Attachment is automatically added to the mail object after creation.
+     *
+     * See also: http://framework.zend.com/manual/1.12/de/zend.mail.attachments.html
+     *
+     * @param  string $body
+     * @param  string $mimeType
+     * @param  string $disposition
+     * @param  string $encoding
+     * @param  string $filename A filename for the attachment
+     * @return $this
+     */
+    public function addAttachment(
+        $body,
+        $mimeType = \Zend_Mime::TYPE_OCTETSTREAM,
+        $disposition = \Zend_Mime::DISPOSITION_ATTACHMENT,
+        $encoding = \Zend_Mime::ENCODING_BASE64,
+        $filename = null
+    ) {
+        $this->message->createAttachment($body, $mimeType, $disposition, $encoding, $filename);
```

Comment:
> We can not add it now as that will be backward incompatible change

First, I do not understand why this should be a backward incompatible change due to the only extensible kind of change and no change on the existing code (new method, no override).

Second, the internal message entity attribute is refering to the Magento2 Message entity and **not** to the Message interface, which is for me a bug itself.
The Magento2 Message entity is extending from the Zend_Mail entity. 

I do not see any conflict with Magento2 architecture concept or backward compatibilities.